website/servers/manage.php

```diff
@@ -0,0 +1,344 @@
+	<div class="ladder pad"><table>
+	<tr><th>Search servers</th><th>Search server modlog</th>
+	<th>Blacklist users from owning servers</th><th>Ban hosting methods</th>
+	<tr><td style="vertical-align: top; max-width: 300px">
+	<form action="/servers/manage.php" data-target="push">
+		<label for="hostedby">Search by host:</label><br />
+		<input type="text" name="hostedby" class="textbox" /><br /><br />
+		<label for="ownedby">Search by owner:</label><br />
+		<input type="text" name="ownedby" class="textbox" /><br /><br />
+		<label for="lastactive">Search by inactivity since a date:</label><br />
+		<input type="date" name="lastactive" class="textbox" /><br /><br />
+		<button type="submit">Search</button>
+	</form>
+	</div>
+	</td><td style="vertical-align: top; max-width: 300px">
+	<form action="/servers/manage.php" data-target="push">
+		<label for="serverid"> Search on a server:</label><br />
+		<input type="text" name="serverid" class="textbox" /><br /><br />
+		<label for="action">Search an action type:</label><br />
+		<input type="text" name="action" class="textbox" /><br /><br />
+		<label for="userid">Search for actions by a userid:</label><br />
+		<input type="text" name="userid" class="textbox" /><br /><br />
+		<button type="submit">Search</button>
+	</form>
+	</div></td><td style="vertical-align: top; max-width: 300px">
+	<form action="/servers/manage.php" data-target="push">
+		<label>(Separate userids with commas)</label><br />
+		<input type="text" name="banowners" class="textbox" /><br /><br />
+		<button type="submit">Add users to list</button>
+	</form>
+<?php
+
+if (isset($_REQUEST['banowners'])) {
+	echo('<br />');
+	$banned = explode(',', $_REQUEST['banowners']);
+	$successes = 0;
+	foreach ($banned as $user) {
+		$userid = $users->userid($user);
+		if (!mb_strlen($userid)) continue;
+		if (!isset($serverinfo['blacklist'])) {
+			$serverinfo['blacklist'] = [];
+		}
+		$failures = [];
+		if (!in_array($userid, $serverinfo['blacklist'])) {
+			$serverinfo['blacklist'][] = $userid;
+			$successes += 1;
+			$psdb->query(
+				"INSERT INTO `{$psdb->prefix}usermodlog` (`userid`,`actorid`,`date`,`ip`,`entry`) VALUES (?, ?, ?, ?, ?)",
+				[$userid, $curuser['userid'], time(), $users->getIp(), 'banned from owning servers']
+			);
+			foreach ($PokemonServers as $server) {
+				if ($server['owner'] == $userid) {
+					unset($PokemonServers[$server['id']]['owner']);
+				}
+				saveservers();
+				$psdb->query(
+					"INSERT INTO {$psdb->prefix}servermodlog (`serverid`, `actorid`, `date`, `ip`, `type`, `note`) VALUES (?, ?, ?, ?, ?, ?)",
+					[$server['id'], $curuser['id'], time(), $users->getIp(), 'BANOWNER', '']
+				);
+			}
+		} else {
+			$failures[] = $userid;
+		}
+		writeserverinfo();
+	}
+	echo "<strong>{$successes} user (s) added to blacklist.</strong><br />";
+	if (count($failures) > 0) {
+		echo ("(" . count($failures) . "user (s) could not be added: " . implode(", ", $failures) . ")<br />");
```

Comment:
```suggestion
		echo "(" . count($failures) . "user (s) could not be added: " . implode(", ", $failures) . ")<br />";
```

`echo` isn't a function.